CONTRIBUTING.md

```diff
@@ -578,6 +580,29 @@ local max_len = 100
 local MAX_LEN = 100
 ```
 
+Minimize the scope of the variables or functions,
+if some module-level variables or functions are used by only one function,
+**do** put them into a `do ... end` block.
+
+```lua
+-- bad
+local MAX_LEN = 100   -- only be used by one function
+
+local function validate()
+
+end
+
+-- good
+local validate
+do
+  local MAX_LEN = 100
+
+  validate = function()
+
+  end
+end
+```
+
```

Comment:
What is the purpose of this rule?  Most of the time, everything that is configurable should live in the beginning of the file.

Reply:
This will limit the visibility of MAX_LEN and make it more secure.

Reply:
Yes, In the Kong repo we already did this many times, it is very common now.

Reply:
> This will limit the visibility of MAX_LEN and make it more secure.

What do you mean by "make it more secure"?

I have been asked to move global configuration variables like these to the top of files several times, so I wonder why we establish a new rule that is contrary to such reviews and to the apparently common understanding.

Reply:
I would rather see people split out modules into separate files than add artficial scopes.

> 1 more thing™️: the linter rules for shadowing variables have been disabled afaik, because they are very annoying, especially with error handling.

I usually fix these issues when I see them because my IDE warns me about them.